lib/foreman/silenced_logger.rb

```diff
@@ -1,16 +1,51 @@
 # Wrap a ::Logging::Logger and implement #silence to temporarily increase the
 # min log level to execute a block of code. Used by sprockets-rails' quiet
-# assets logging feature.
+# assets logging feature. Foreman uses logging gem instead Rails logging stack
+# and it does not provide silencing feature (ships with just a noop method).
+# This implementation only works with Logging gem, not with Ruby/Rails Logger.
+#
+# Inspired by lib/active_record/session_store/extension/logger_silencer.rb
+#
 module Foreman
   class SilencedLogger < SimpleDelegator
-    def silence(new_level = Logger::ERROR, &block)
-      old_level = self.level
-      begin
-        self.level = new_level
-        yield self
-      ensure
-        self.level = old_level
+    def initialize(obj)
+      ::Logging::LEVELS.each do |name, num|
```

Comment:
The metaprogramming part should not be in the initializer, but rather outside of any method: otherwise we would loose method lookup cache everytime we initialize the silencer.

Given you override `log_event` now, do we even need to override this methods?

Reply:
Putting this outside of initializer does not work, it simply does not create the methods correctly. If you show me a way I'd happily do that.

Overriding `log_event` is actually useless, I will remove that. The thing is this is a decorator, not superclass. The only way I found was to actually explicitly create methods.